daktari/checks/kubernetes.py

```diff
@@ -55,3 +56,60 @@ def check(self) -> CheckResult:
         output = get_stdout("kubectl config get-contexts")
         passed = bool(output and self.context_name in output)
         return self.verify(passed, f"{self.context_name} is <not/> configured for the current user")
+
+
+class HelmInstalled(Check):
+    def __init__(self, minimum_version: Optional[float] = None):
+        self.minimum_version = minimum_version
+        self.name = "Helm.installed"
+        self.suggestions = {
+            OS.OS_X: "<cmd>brew install helm</cmd>",
+            OS.UBUNTU: "<cmd>sudo snap install helm --classic</cmd>",
+            OS.GENERIC: "Install Helm: https://helm.sh/docs/intro/install/",
+        }
+
+    def check(self) -> CheckResult:
+        installed_version = get_helm_version()
+        if installed_version is None:
+            return self.failed("Helm is not installed")
+
+        if self.minimum_version is None:
+            return self.passed("Helm is installed")
+
+        return self.verify(
+            installed_version >= self.minimum_version,
+            f"Helm version is <not/> >={self.minimum_version} ({installed_version})",
+        )
```

Comment:
This is the third time we're writing this kind of flow - could we extract a helper for it? Obviously actually getting the version will be bespoke each time (different commands / regex), but from there I think it's all common - you'd just need to pass the name of the thing (`Helm`, `Docker` or `Kubectl`)

Reply:
I was thinking of not including the regex part, although that is a bit duplicatey. But something that would return a `CheckResult` where you'd call like:

`return validate_version("Helm", version, minimum_version)`

And it would do the "is installed", "not installed", "not high enough version" branches.

Reply:
I'm not sure if that's possible though, considering the different commands and different outputs. 

i.e. 
`helm version --short`
`kubectl version --client=true --short` 

They're common in that they have that kind of structure (`<app> version --short`), but if we don't pass --client to kubectl version for example it'll return the version number for the kubectl context it's connected to and I'm not sure how we'd write a common regex check for that to match the version string etc. 

```
$ kubectl version --short 
Client Version: v1.18.3
Server Version: v1.20.6-gke.1000
```

Whereas if we pass in the command, and regex that feels more universal and reusable? Or am I really going down a rabbit hole here 😂

Reply:
As a first step, to keep it simple, I was thinking that you would pass the ultimately parsed `float`. So the Helm example would end up like:

```
installed_version = get_helm_version()
return validate_version("Helm", version, minimum_version)
```

I.e. keep the command, regex, converting to float out of it for now, implemented individually, and just reuse from the point where you have a (possibly undefined) numeric version to work with

Reply:
Ah! I'm following now, I've pushed 📌